test/ensembl/ensembl.uploadFile.test.js

```diff
@@ -0,0 +1,256 @@
+const fs = require('fs');
+const path = require('path');
+const fetchMock = require('node-fetch');
+const { uploadFile } = require('../../src/ensembl');
+const { ApiConnection } = require('../../src/graphkb');
+
+
+// Mock node-fetch calls to external APIs
+jest.mock('node-fetch', () => require('fetch-mock-jest').sandbox()); // eslint-disable-line global-require
+
+const extApiData = JSON.parse(fs.readFileSync(path.join(__dirname, '../data/external_api_responses.json')));
+Object.entries(extApiData).forEach(([url, respBody]) => {
+    fetchMock.get(url, respBody);
+});
+
+// Mock ApiConnexion class with overrided request function
+class MockApiConnection extends ApiConnection {
+    constructor() {
+        super();
+        // Data in mockfile are ordered to match actual request order
+        this.mockFile = path.join(__dirname, '../data/ensembl_uploadFile_requests.json');
+        this.mockDataset = JSON.parse(fs.readFileSync(this.mockFile));
+    }
+
+    request() {
+        const { result } = this.mockDataset[0];
+        this.mockDataset.shift();
```

Comment:
i believe shift() returns array[0] so you can just do const { result } = this.mockDataset.shift()

Reply:
yes, totally. I'll change that